config/v2/config.go

```diff
@@ -152,12 +152,18 @@ type DatadogProvider struct{}
 
 //Backend is used to configure a terraform backend
 type Backend struct {
-	Type        *string `json:"type,omitempty" yaml:"type,omitempty" validate:"omitempty,oneof=s3"`
+	Kind *string `json:"kind,omitempty" yaml:"kind,omitempty" validate:"omitempty,oneof=s3 remote"`
```

Comment:
I see we have a type alias to string for `BackendKind` in plan, worth using it here too?

Reply:
👍

Reply:
I like the idea, but my only problem with that invalid backend types would fail (panic?) at parsing time rather than validation. I think making that check part of validations makes for a better user experience.